### PR TITLE
fix: add support for nixbuilds for ecs fargate

### PR DESCRIPTION
**Pull request by boetro**
## Testing

Automated testing, and made sure all combinations of building worked

### CODE REVIEW

File: launchflow/builds/ecr_builder.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        self.build_directory = build_directory
+        self.build_ignore = build_ignore
+        self.build_log_file = build_log_file
+        self.ecr_repository = ecr_repository
+        self.launchflow_project_name = launchflow_project_name
+        self.launchflow_environment_name = launchflow_environment_name
+        self.launchflow_service_name = launchflow_service_name
+        self.launchflow_deployment_id = launchflow_deployment_id
+        self.aws_environment_config = aws_environment_config
+        self.docker_client = from_env()
```
Comment (JoshTanke) [inline launchflow/builds/ecr_builder.py:68]: IIRC we wrap the other calls to from_env in a try / catch so we can raise a special Docker not installed error

File: launchflow/cli/main.py [hunk 2]
```diff
@@ -120,6 +120,12 @@ def _set_global_project_and_environment(
 """
 
 
+# Flow should be
+# Detect all frameworks
+# Prompt user to select their cloud provider
+# Prompt user to select a frameworks to deploy
+# Prompt user to select runtime
+# Prompt user to select build
```
Comment (JoshTanke) [inline launchflow/cli/main.py:128]: did you mean to leave these comments? aside: we should sketch out the ideal cli flow on a doc / maybe stumble it before we start coding it up

**Review by JoshTanke — APPROVED** [review]

File: launchflow/builds/ecr_builder.py (hunk 1 above)
Comment (boetro) [reply launchflow/builds/ecr_builder.py:68]: docker is in the pyproject.toml so should be fine: https://github.com/launchflow/launchflow/blob/a8cbd3a4989146527d13c9d11a5fab4ba85f140e/pyproject.toml#L26

File: launchflow/cli/main.py (hunk 2 above)
Comment (boetro) [reply launchflow/cli/main.py:128]: yea I just left them here for reference from what we talked about

**Pull request merged**